element_zstack/bossdb.py

```diff
@@ -0,0 +1,161 @@
+import importlib
+import inspect
+import logging
+
+import datajoint as dj
+import numpy as np
+from element_zstack import volume
+
+from .export.bossdb_interface import BossDBUpload
+
+logger = logging.getLogger("datajoint")
+
+schema = dj.Schema()
+_linking_module = None
+
+
+def activate(
+    schema_name: str,
+    *,
+    create_schema: bool = True,
+    create_tables: bool = True,
+    linking_module: str = None,
+):
+    """Activate this schema
+
+    Args:
+        schema_name (str): schema name on the database server to activate the `bossdb` schema
+        create_schema (bool): when True (default), create schema in the database if it
+                            does not yet exist.
+        create_tables (bool): when True (default), create schema tables in the database
+                             if they do not yet exist.
+        linking_module (str): A string containing the module name or module containing
+            the required dependencies to activate the schema.
+
+    Dependencies:
+    Tables:
+        volume.Volume: A parent table to VolumeUploadTask
+        volume.VoxelSize: A dependency of BossDBURLs
+    """
+
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'linking_module' must be a module's name or a module"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    schema.activate(
+        schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
+
+
+@schema
+class VolumeUploadTask(dj.Manual):
+    """A pairing of Volume data to be uploaded and parameter set to be used for
+    the upload.
+
+    Attributes:
+        volume.Volume (foreign key): Primary key from `Volume`.
+        upload_type (str): One of 'image' (volumetric image) or 'annotation'
+        (segmentation data).
+        collection_name (varchar(64)): Name of the collection on bossdb.
+        experiment_name (varchar(64)): Name of the experiment on bossdb.
+        channel_name (varchar(64)): Name of the channel on bossdb.
+    """
+
+    definition = """
+    -> volume.Volume
+    upload_type='image': enum('image', 'annotation')
```

Comment:
I might be misunderstanding, but is it not possible for the Neuroglancer link to view the volumetric image and the segmented data in the same view?

Reply:
Yes, Neuroglancer links are highly flexible and customizable, depending on what the viewer wants to see in their image. There is a way to load segmented data from within the Neuroglancer webapp, and this would generate a different URL. This implementation of adding a single neuroglancer link pointing to a single upload is the least likely to result in bugs and leaves the user with the option to just upload a volumetric image and view it without requiring the segmentation data.

Reply:
Thanks for clarifying.  We should automate the Neuroglancer view to show the image and/or the segmentation.  We can update the `VolumeUploadTask` so that a user can specify if they would like to upload the image, segmentation, or both.  Otherwise we will need to provide instructions for a user to load the segmented data from within the Neuroglancer webapp.  What kind of bugs have you encountered when creating a link to view both the image and segmentation?

Reply:
Based on offline code review and discussion, the `VolumeUploadTask` and `VolumeUpload` tables have been updated. The new make function can now be reviewed